core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -116,6 +123,16 @@ public void run() {
     public void shutdown() {
         shuttingDown = true;
         try {
+            SocketAddress localAddress = serverSocket.getLocalAddress();
```

Comment:
Well if I were writing this from scratch, yep... but that's not how `AgentProtocol` works :cry: